## DATA LOADING

In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('DataFiles/Cleaned_unemployed.csv')

In [3]:
df

country_name                    indicator_name     sex age_group  \
0     Afghanistan  Unemployment rate by sex and age  Female     15-24   
1     Afghanistan  Unemployment rate by sex and age  Female       25+   
2     Afghanistan  Unemployment rate by sex and age  Female  Under 15   
3     Afghanistan  Unemployment rate by sex and age    Male     15-24   
4     Afghanistan  Unemployment rate by sex and age    Male       25+   
...           ...                               ...     ...       ...   
1129     Zimbabwe  Unemployment rate by sex and age  Female       25+   
1130     Zimbabwe  Unemployment rate by sex and age  Female  Under 15   
1131     Zimbabwe  Unemployment rate by sex and age    Male     15-24   
1132     Zimbabwe  Unemployment rate by sex and age    Male       25+   
1133     Zimbabwe  Unemployment rate by sex and age    Male  Under 15   

     age_categories    2014    2015    2016    2017    2018    2019    2020  \
0             Youth  13.340  15.974  18.570  21.137  20.649  20.154  21.228   
1            Adults   8.576   9.014   9.463   9.920  11.223  12.587  14.079   
2          Children  10.306  11.552  12.789  14.017  14.706  15.418  16.783   
3             Youth   9.206  11.502  13.772  16.027  15.199  14.361  14.452   
4            Adults   6.463   6.879   7.301   7.728   7.833   7.961   8.732   
...             ...     ...     ...     ...     ...     ...     ...     ...   
1129         Adults   4.262   4.586   4.773   4.900   5.027   5.309   6.345   
1130       Children   5.174   5.666   6.014   6.289   6.555   6.990   8.494   
1131          Youth   6.007   7.131   8.119   9.029   9.927  11.048  13.191   
1132         Adults   3.677   4.325   4.843   5.287   5.722   6.362   6.996   
1133       Children   4.355   5.152   5.821   6.410   6.984   7.760   8.809   

        2021    2022    2023    2024  
0     21.640  30.561  32.200  33.332  
1     14.415  23.818  26.192  28.298  
2     17.134  26.746  29.193  30.956  
3     15.099  16.655  18.512  19.770  
4      9.199  11.357  12.327  13.087  
...      ...     ...     ...     ...  
1129   7.792   7.704   7.597   7.528  
1130   9.853   9.670   9.518   9.414  
1131  14.261  13.966  13.675  13.485  
1132   7.234   6.858   6.757   6.703  
1133   9.223   8.842   8.713   8.632  

[1134 rows x 16 columns]

## DATA CLEANING

In [4]:
df.isnull().sum()

country_name      0
indicator_name    0
sex               0
age_group         0
age_categories    0
2014              0
2015              0
2016              0
2017              0
2018              0
2019              0
2020              0
2021              0
2022              0
2023              0
2024              0
dtype: int64

In [5]:
df[df.duplicated()]

Empty DataFrame
Columns: [country_name, indicator_name, sex, age_group, age_categories, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023, 2024]
Index: []

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1134 entries, 0 to 1133
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   country_name    1134 non-null   object 
 1   indicator_name  1134 non-null   object 
 2   sex             1134 non-null   object 
 3   age_group       1134 non-null   object 
 4   age_categories  1134 non-null   object 
 5   2014            1134 non-null   float64
 6   2015            1134 non-null   float64
 7   2016            1134 non-null   float64
 8   2017            1134 non-null   float64
 9   2018            1134 non-null   float64
 10  2019            1134 non-null   float64
 11  2020            1134 non-null   float64
 12  2021            1134 non-null   float64
 13  2022            1134 non-null   float64
 14  2023            1134 non-null   float64
 15  2024            1134 non-null   float64
dtypes: float64(11), object(5)
memory usage: 141.9+ KB


## DATA MANIPULATION

## The Countries Listed

In [7]:
dfCountriesFiltered = df['country_name'].unique()
dfCountriesFiltered

array(['Afghanistan', 'Albania', 'Algeria', 'Angola', 'Argentina',
       'Armenia', 'Australia', 'Austria', 'Azerbaijan', 'Bahamas',
       'Bahrain', 'Bangladesh', 'Barbados', 'Belarus', 'Belgium',
       'Belize', 'Benin', 'Bhutan', 'Bolivia', 'Bosnia and Herzegovina',
       'Botswana', 'Brazil', 'Brunei Darussalam', 'Bulgaria',
       'Burkina Faso', 'Burundi', 'Cabo Verde', 'Cambodia', 'Cameroon',
       'Canada', 'Central African Republic', 'Chad', 'Channel Islands',
       'Chile', 'China', 'Colombia', 'Comoros', 'Congo',
       'Congo, Democratic Republic of the', 'Costa Rica', 'Croatia',
       'Cuba', 'Cyprus', 'Czechia', 'Ivory Coast', 'Denmark', 'Djibouti',
       'Dominican Republic', 'Ecuador', 'Egypt', 'El Salvador',
       'Equatorial Guinea', 'Eritrea', 'Estonia', 'Eswatini', 'Ethiopia',
       'Fiji', 'Finland', 'France', 'French Polynesia', 'Gabon', 'Gambia',
       'Georgia', 'Germany', 'Ghana', 'Greece', 'Guam', 'Guatemala',
       'Guinea', 'Guinea-Bissau', 'Guya

The Countries whose statistics in regard to unemployment were used are as listed above.

## Averages per year with respect to age group, sex and both age group & sex

2014

In [8]:
average2014 = df.groupby(['sex','age_categories'])['2014'].mean().reset_index()
average2014

sex age_categories       2014
0  Female         Adults   7.566878
1  Female       Children   9.521116
2  Female          Youth  20.435132
3    Male         Adults   5.922524
4    Male       Children   7.608074
5    Male          Youth  17.273074

The youth who are female have the largest average at 20.44 followed by their male counterparts at 17.27. The least average belongs to the male adults at 5.92

In [9]:
average2014Sex = df.groupby(['sex'])['2014'].mean().reset_index()
average2014Sex

sex       2014
0  Female  12.507709
1    Male  10.267891

Females have the largest at 12.51 while the males follow at 10.27

In [10]:
average2014AgeGroup = df.groupby(['age_categories'])['2014'].mean().reset_index()
average2014AgeGroup

age_categories       2014
0         Adults   6.744701
1       Children   8.564595
2          Youth  18.854103

The youth have the largest rate folowed by children then adults. There is a considerable difference between the first and second

2015

In [11]:
average2015 = df.groupby(['sex','age_categories'])['2015'].mean().reset_index()
average2015

sex age_categories       2015
0  Female         Adults   7.543069
1  Female       Children   9.459386
2  Female          Youth  20.237407
3    Male         Adults   5.863529
4    Male       Children   7.509635
5    Male          Youth  17.021640

The female youth still remain at the top followed by their male counterparts, and the bottom being occupied by male adults

In [12]:
average2015Sex = df.groupby(['sex'])['2015'].mean().reset_index()
average2015Sex

sex       2015
0  Female  12.413287
1    Male  10.131601

The females stiill lead the males

In [13]:
average2015AgeGroup = df.groupby(['age_categories'])['2015'].mean().reset_index()
average2015AgeGroup

age_categories       2015
0         Adults   6.703299
1       Children   8.484511
2          Youth  18.629524

The youth still lead, with their gap with respect to the children still considerably large

2016

In [14]:
average2016 = df.groupby(['sex','age_categories'])['2016'].mean().reset_index()
average2016

sex age_categories       2016
0  Female         Adults   7.446280
1  Female       Children   9.336857
2  Female          Youth  20.091085
3    Male         Adults   5.764254
4    Male       Children   7.374212
5    Male          Youth  16.725090

In [15]:
average2016Sex = df.groupby(['sex'])['2016'].mean().reset_index()
average2016Sex

sex       2016
0  Female  12.291407
1    Male   9.954519

In [16]:
average2016AgeGroup = df.groupby(['age_categories'])['2016'].mean().reset_index()
average2016AgeGroup

age_categories       2016
0         Adults   6.605267
1       Children   8.355534
2          Youth  18.408087

The trend remains the same as the previous year

2017

In [17]:
average2017 = df.groupby(['sex','age_categories'])['2017'].mean().reset_index()
average2017

sex age_categories       2017
0  Female         Adults   7.312688
1  Female       Children   9.149899
2  Female          Youth  19.828688
3    Male         Adults   5.567730
4    Male       Children   7.117984
5    Male          Youth  16.204106

In [18]:
average2017Sex = df.groupby(['sex'])['2017'].mean().reset_index()
average2017Sex

sex       2017
0  Female  12.097092
1    Male   9.629940

In [19]:
average2017AgeGroup = df.groupby(['age_categories'])['2017'].mean().reset_index()
average2017AgeGroup

age_categories       2017
0         Adults   6.440209
1       Children   8.133942
2          Youth  18.016397

The trend remains the same as the previous year

2018

In [20]:
average2018 = df.groupby(['sex','age_categories'])['2018'].mean().reset_index()
average2018

sex age_categories       2018
0  Female         Adults   7.040540
1  Female       Children   8.800593
2  Female          Youth  19.279847
3    Male         Adults   5.362407
4    Male       Children   6.860762
5    Male          Youth  15.754847

In [21]:
average2018Sex = df.groupby(['sex'])['2018'].mean().reset_index()
average2018Sex

sex       2018
0  Female  11.706993
1    Male   9.326005

In [22]:
average2018AgeGroup = df.groupby(['age_categories'])['2018'].mean().reset_index()
average2018AgeGroup

age_categories       2018
0         Adults   6.201474
1       Children   7.830677
2          Youth  17.517347

The trend remains the same as the previous year

2019

In [23]:
average2019 = df.groupby(['sex','age_categories'])['2019'].mean().reset_index()
average2019

sex age_categories       2019
0  Female         Adults   6.998122
1  Female       Children   8.696021
2  Female          Youth  18.919730
3    Male         Adults   5.208979
4    Male       Children   6.665026
5    Male          Youth  15.380836

In [24]:
average2019Sex = df.groupby(['sex'])['2019'].mean().reset_index()
average2019Sex

sex       2019
0  Female  11.537958
1    Male   9.084947

In [25]:
average2019AgeGroup = df.groupby(['age_categories'])['2019'].mean().reset_index()
average2019AgeGroup

age_categories       2019
0         Adults   6.103550
1       Children   7.680524
2          Youth  17.150283

The trend remains the same as the previous year

2020

In [26]:
average2020 = df.groupby(['sex','age_categories'])['2020'].mean().reset_index()
average2020

sex age_categories       2020
0  Female         Adults   7.795714
1  Female       Children   9.739407
2  Female          Youth  21.932302
3    Male         Adults   6.190683
4    Male       Children   7.765942
5    Male          Youth  17.683661

In [27]:
average2020Sex = df.groupby(['sex'])['2020'].mean().reset_index()
average2020Sex

sex       2020
0  Female  13.155808
1    Male  10.546762

In [28]:
average2020AgeGroup = df.groupby(['age_categories'])['2020'].mean().reset_index()
average2020AgeGroup

age_categories       2020
0         Adults   6.993198
1       Children   8.752675
2          Youth  19.807981

The trend remains the same as the previous year

2021

In [29]:
average2021 = df.groupby(['sex','age_categories'])['2021'].mean().reset_index()
average2021

sex age_categories       2021
0  Female         Adults   7.750799
1  Female       Children   9.522381
2  Female          Youth  20.848032
3    Male         Adults   6.001286
4    Male       Children   7.492106
5    Male          Youth  16.921265

In [30]:
average2021Sex = df.groupby(['sex'])['2021'].mean().reset_index()
average2021Sex

sex       2021
0  Female  12.707071
1    Male  10.138219

In [31]:
average2021AgeGroup = df.groupby(['age_categories'])['2021'].mean().reset_index()
average2021AgeGroup

age_categories       2021
0         Adults   6.876042
1       Children   8.507243
2          Youth  18.884648

The trend remains the same

2022

In [32]:
average2022 = df.groupby(['sex','age_categories'])['2022'].mean().reset_index()
average2022

sex age_categories       2022
0  Female         Adults   7.086613
1  Female       Children   8.699909
2  Female          Youth  19.016441
3    Male         Adults   5.270994
4    Male       Children   6.642845
5    Male          Youth  15.392993

In [33]:
average2022Sex = df.groupby(['sex'])['2022'].mean().reset_index()
average2022Sex

sex       2022
0  Female  11.600988
1    Male   9.102277

In [34]:
average2022AgeGroup = df.groupby(['age_categories'])['2022'].mean().reset_index()
average2022AgeGroup

age_categories       2022
0         Adults   6.178804
1       Children   7.671377
2          Youth  17.204717

The trend is still the same

2023

In [35]:
average2023 = df.groupby(['sex','age_categories'])['2023'].mean().reset_index()
average2023

sex age_categories       2023
0  Female         Adults   6.924269
1  Female       Children   8.500808
2  Female          Youth  18.527637
3    Male         Adults   5.132553
4    Male       Children   6.503724
5    Male          Youth  15.173408

In [36]:
average2023Sex = df.groupby(['sex'])['2023'].mean().reset_index()
average2023Sex

sex       2023
0  Female  11.317571
1    Male   8.936562

In [37]:
average2023AgeGroup = df.groupby(['age_categories'])['2023'].mean().reset_index()
average2023AgeGroup

age_categories       2023
0         Adults   6.028411
1       Children   7.502266
2          Youth  16.850523

The trend is still the same

2024

In [38]:
average2024 = df.groupby(['sex','age_categories'])['2024'].mean().reset_index()
average2024

sex age_categories       2024
0  Female         Adults   6.903312
1  Female       Children   8.466776
2  Female          Youth  18.524325
3    Male         Adults   5.066109
4    Male       Children   6.433893
5    Male          Youth  15.100297

In [39]:
average2024Sex = df.groupby(['sex'])['2024'].mean().reset_index()
average2024Sex

sex       2024
0  Female  11.298138
1    Male   8.866766

In [40]:
average2024AgeGroup = df.groupby(['age_categories'])['2024'].mean().reset_index()
average2024AgeGroup

age_categories       2024
0         Adults   5.984710
1       Children   7.450334
2          Youth  16.812311

The trend remains the same

## Global Average Between 2014 and 2024

The averages for both sex and age groups can be viewed from the following table;

In [70]:
averageAgeGroupSex={
    'sex': ['Female', 'Female', 'Female', 'Male', 'Male', 'Male'],
    'age_categories': ['Adults', 'Children', 'Youth', 'Adults', 'Children', 'Youth'],
    '2014': [7.566878,9.521116,20.435132,5.922524,7.608074,17.273074],
    '2015': [7.543069,9.459386,20.237407,5.863529,7.509635,17.021640],
    '2016': [7.446280,9.336857,20.091085,5.764254,7.374212,16.725090],
    '2017': [7.312688,9.149899,19.828688,5.567730,7.117984,16.204106],
    '2018': [7.040540,8.800593,19.279847,5.362407,6.860762,15.754847],
    '2019': [6.998122,8.696021,18.919730,5.208979,6.665026,15.380836],
    '2020': [7.795714,9.739407,21.932302,6.190683,7.765942,17.683661],
    '2021': [7.750799,9.522381,20.848032,6.001286,7.492106,16.921265],
    '2022': [7.086613,8.699909,19.016441,5.270994,6.642845,15.392993],
    '2023': [6.924269,8.500808,18.527637,5.132553,6.503724,15.173408],
    '2024': [6.903312,8.466776,18.524325,5.066109,6.433893,15.100297]
}

In [72]:
dfAverageAgeGroupSex=pd.DataFrame(averageAgeGroupSex)
dfAverageAgeGroupSex

sex age_categories       2014       2015       2016       2017  \
0  Female         Adults   7.566878   7.543069   7.446280   7.312688   
1  Female       Children   9.521116   9.459386   9.336857   9.149899   
2  Female          Youth  20.435132  20.237407  20.091085  19.828688   
3    Male         Adults   5.922524   5.863529   5.764254   5.567730   
4    Male       Children   7.608074   7.509635   7.374212   7.117984   
5    Male          Youth  17.273074  17.021640  16.725090  16.204106   

        2018       2019       2020       2021       2022       2023       2024  
0   7.040540   6.998122   7.795714   7.750799   7.086613   6.924269   6.903312  
1   8.800593   8.696021   9.739407   9.522381   8.699909   8.500808   8.466776  
2  19.279847  18.919730  21.932302  20.848032  19.016441  18.527637  18.524325  
3   5.362407   5.208979   6.190683   6.001286   5.270994   5.132553   5.066109  
4   6.860762   6.665026   7.765942   7.492106   6.642845   6.503724   6.433893  
5  15.754847  15.380836  17.683661  16.921265  15.392993  15.173408  15.100297

In [76]:
years= [str(year) for year in range(2014, 2024 + 1)]
dfAverageAgeGroupSex['average'] = dfAverageAgeGroupSex[years].mean(axis=1)
dfAverageAgeGroupSex

sex age_categories       2014       2015       2016       2017  \
0  Female         Adults   7.566878   7.543069   7.446280   7.312688   
1  Female       Children   9.521116   9.459386   9.336857   9.149899   
2  Female          Youth  20.435132  20.237407  20.091085  19.828688   
3    Male         Adults   5.922524   5.863529   5.764254   5.567730   
4    Male       Children   7.608074   7.509635   7.374212   7.117984   
5    Male          Youth  17.273074  17.021640  16.725090  16.204106   

        2018       2019       2020       2021       2022       2023  \
0   7.040540   6.998122   7.795714   7.750799   7.086613   6.924269   
1   8.800593   8.696021   9.739407   9.522381   8.699909   8.500808   
2  19.279847  18.919730  21.932302  20.848032  19.016441  18.527637   
3   5.362407   5.208979   6.190683   6.001286   5.270994   5.132553   
4   6.860762   6.665026   7.765942   7.492106   6.642845   6.503724   
5  15.754847  15.380836  17.683661  16.921265  15.392993  15.173408   

        2024    average  
0   6.903312   7.306208  
1   8.466776   9.081196  
2  18.524325  19.785511  
3   5.066109   5.577368  
4   6.433893   7.088564  
5  15.100297  16.239202

In [78]:
dfAverageAgeGroupSexSelected = dfAverageAgeGroupSex[['sex', 'age_categories', 'average']]
dfAverageAgeGroupSexSelected

sex age_categories    average
0  Female         Adults   7.306208
1  Female       Children   9.081196
2  Female          Youth  19.785511
3    Male         Adults   5.577368
4    Male       Children   7.088564
5    Male          Youth  16.239202

The listed above averages represent the unemployment rate from 2014 to 2024 with respect to both sex and age category

In [80]:
averageSex={
    'sex': ['Female', 'Male'],
    '2014': [12.507709,10.267891],
    '2015': [12.413287,10.131601],
    '2016': [12.291407,9.954519],
    '2017': [12.097092,9.629940],
    '2018': [11.706993,9.326005],
    '2019': [11.537958,9.084947],
    '2020': [13.155808,10.546762],
    '2021': [12.707071,10.138219],
    '2022': [11.600988,9.102277],
    '2023': [11.317571,8.936562],
    '2024': [11.298138,8.866766]
}

In [81]:
dfAverageSex=pd.DataFrame(averageSex)
dfAverageSex

sex       2014       2015       2016       2017       2018       2019  \
0  Female  12.507709  12.413287  12.291407  12.097092  11.706993  11.537958   
1    Male  10.267891  10.131601   9.954519   9.629940   9.326005   9.084947   

        2020       2021       2022       2023       2024  
0  13.155808  12.707071  11.600988  11.317571  11.298138  
1  10.546762  10.138219   9.102277   8.936562   8.866766

In [82]:
yearsSex= [str(yearx) for yearx in range(2014, 2024 + 1)]
dfAverageSex['average'] = dfAverageSex[yearsSex].mean(axis=1)
dfAverageSex

sex       2014       2015       2016       2017       2018       2019  \
0  Female  12.507709  12.413287  12.291407  12.097092  11.706993  11.537958   
1    Male  10.267891  10.131601   9.954519   9.629940   9.326005   9.084947   

        2020       2021       2022       2023       2024    average  
0  13.155808  12.707071  11.600988  11.317571  11.298138  12.057638  
1  10.546762  10.138219   9.102277   8.936562   8.866766   9.635044

In [83]:
dfAverageSexSelected = dfAverageSex[['sex', 'average']]
dfAverageSexSelected

sex    average
0  Female  12.057638
1    Male   9.635044

The above averages represent the unemployment rate according to gender from 2014 to 2024

In [84]:
averageAgeGroup={
    'age_categories': ['Adults', 'Children','Youth'],
    '2014': [6.744701,8.564595,18.854103],
    '2015': [6.703299,8.484511,18.629524],
    '2016': [6.605267,8.355534,18.408087],
    '2017': [6.440209,8.133942,18.016397],
    '2018': [6.201474,7.830677,17.517347],
    '2019': [6.103550,7.680524,17.150283],
    '2020': [6.993198,8.752675,19.807981],
    '2021': [6.876042,8.507243,18.884648],
    '2022': [6.178804,7.671377,17.204717],
    '2023': [6.028411,7.502266,16.850523],
    '2024': [5.984710,7.450334,16.812311]
}

In [85]:
dfAverageAgeGroup=pd.DataFrame(averageAgeGroup)
dfAverageAgeGroup

age_categories       2014       2015       2016       2017       2018  \
0         Adults   6.744701   6.703299   6.605267   6.440209   6.201474   
1       Children   8.564595   8.484511   8.355534   8.133942   7.830677   
2          Youth  18.854103  18.629524  18.408087  18.016397  17.517347   

        2019       2020       2021       2022       2023       2024  
0   6.103550   6.993198   6.876042   6.178804   6.028411   5.984710  
1   7.680524   8.752675   8.507243   7.671377   7.502266   7.450334  
2  17.150283  19.807981  18.884648  17.204717  16.850523  16.812311

In [86]:
yearsAgeGroup= [str(yearg) for yearg in range(2014, 2024 + 1)]
dfAverageAgeGroup['average'] = dfAverageAgeGroup[yearsAgeGroup].mean(axis=1)
dfAverageAgeGroup

age_categories       2014       2015       2016       2017       2018  \
0         Adults   6.744701   6.703299   6.605267   6.440209   6.201474   
1       Children   8.564595   8.484511   8.355534   8.133942   7.830677   
2          Youth  18.854103  18.629524  18.408087  18.016397  17.517347   

        2019       2020       2021       2022       2023       2024    average  
0   6.103550   6.993198   6.876042   6.178804   6.028411   5.984710   6.441788  
1   7.680524   8.752675   8.507243   7.671377   7.502266   7.450334   8.084880  
2  17.150283  19.807981  18.884648  17.204717  16.850523  16.812311  18.012356

In [87]:
dfAverageAgeGroupSelected = dfAverageAgeGroup[['age_categories', 'average']]
dfAverageAgeGroupSelected

age_categories    average
0         Adults   6.441788
1       Children   8.084880
2          Youth  18.012356

The above averages represent the unemployment rate from 2014 to 2024 with reference to the age groups